Doc 13386 (#22)
couchbase/docs-operator: modules/ROOT/partials/couchbase-operator-release-notes-2.8.0.adoc

[#fixed-issues-v280]
=== Fixed Issues

*https://jira.issues.couchbase.com/browse/K8S-3558[K8S-3558^]*::

Operator commences an In-place Upgrade when the cluster is under-resourced.

*https://jira.issues.couchbase.com/browse/K8S-3579[K8S-3579^]*::

Operator tries to change invalid bucket configurations in a loop.

*https://jira.issues.couchbase.com/browse/K8S-3591[K8S-3591^]*::

> Operator crashes if Incremental Backup is missing schedule.

Reviewer: Automated review comment from Vale and OpenAI using the gpt-4o-mini model:

  ```suggestion
  Couchbase Autonomous Operator crashes if Incremental Backup is missing schedule.
  ```
  
  Using these Vale rules:
  * Couchbase.CouchbaseAutoOp (error) - Use the full product name (Couchbase Autonomous Operator) first.

*https://jira.issues.couchbase.com/browse/K8S-3596[K8S-3596^]*::

Crash in Operator due to invalid memory access.

*https://jira.issues.couchbase.com/browse/K8S-3605[K8S-3605^]*::

Upgrade Swap Rebalance is retried with different parameters on Operator Pod deletion.

*https://jira.issues.couchbase.com/browse/K8S-3609[K8S-3609^]*::

 Hibernation fails to bring back any Pod with error extracting image version.

*https://jira.issues.couchbase.com/browse/K8S-3621[K8S-3621^]*::

Shadowed Secret did not get updated.

*https://jira.issues.couchbase.com/browse/K8S-3632[K8S-3632^]*::

Unable to set -1 for Collection-level `maxTTL`.

*https://jira.issues.couchbase.com/browse/K8S-3639[K8S-3639^]*::

Operator loses track of pending Pods when an Eviction of the Operator Pod occurs.

*https://jira.issues.couchbase.com/browse/K8S-3641[K8S-3641^]*::

 Crash in `handleVolumeExpansion` if `enableOnlineVolumeExpansion` is True but no Volume Mounts configured.

*https://jira.issues.couchbase.com/browse/K8S-3655[K8S-3655^]*::

>  Clear Upgrade condition if the Operator isn't performing an upgrade.

Reviewer: Automated review comment from Vale and OpenAI using the gpt-4o-mini model:

  ```suggestion
   Clear Upgrade condition if the Operator is not performing an upgrade.
  ```
  
  Using these Vale rules:
  * Couchbase.NegativeContractions (error) - Use &#x27;is not&#x27; instead of &#x27;isn&#x27;t&#x27;.

*https://jira.issues.couchbase.com/browse/K8S-3659[K8S-3659^]*::

>  When scaling down, Cluster does not maintain balance across Server Groups.

Reviewer: Automated review comment from Vale and OpenAI using the gpt-4o-mini model:

  ```suggestion
   When scaling, Cluster does not maintain balance across Server Groups.
  ```
  
  Using these Vale rules:
  * [Couchbase.DirectionalLanguage](https://docs.couchbase.com/styleguide/directional-language.html) (warning) - Avoid directional language like &#x27;down&#x27;.

Author: Hmm. I think it works in this case and I'll let it slide but maybe there's a better way to say this.

*https://jira.issues.couchbase.com/browse/K8S-3696[K8S-3696^]*::

 DAC prevents configuration of multiple XDCR Replications of same Buckets to different remote Clusters.

*https://jira.issues.couchbase.com/browse/K8S-3772[K8S-3772^]*::

> Self-Certification: Artifacts PVC should use `--storage-class` parameter when creating the Certification Pod.

Reviewer: Automated review comment from Vale and OpenAI using the gpt-4o-mini model:

  ```suggestion
  Self-Certification: artifacts PVC should use `--storage-class` parameter when creating the Certification Pod.
  ```
  
  Using these Vale rules:
  * [Google.Colons](https://developers.google.com/style/colons) (warning) - &#x27;: A&#x27; should be in lowercase.

> Self-Certification: Artifacts PVC should use `--storage-class` parameter when creating the Certification Pod.

Reviewer: Automated review comment from Vale and OpenAI using the gpt-4o-mini model:

  ```suggestion
  Self-Certification: artifacts PVC should use `--storage-class` parameter when creating the Certification Pod.
  ```
  
  Using these Vale rules:
  * [Google.Colons](https://developers.google.com/style/colons) (warning) - &#x27;: A&#x27; should be in lowercase.

*https://jira.issues.couchbase.com/browse/K8S-3788[K8S-3788^]*::

> Operator container crashes when there is a managed Scope/Collection Group added for the Ephemeral Bucket.

Reviewer: Automated review comment from Vale and OpenAI using the gpt-4o-mini model:

  ```suggestion
  Couchbase Autonomous Operator container crashes when there is a managed Scope/Collection Group added for the Ephemeral Bucket.
  ```
  
  Using these Vale rules:
  * Couchbase.CouchbaseAutoOp (error) - Use the full product name (Couchbase Autonomous Operator) first.

|===
| Issue | Description

*https://jira.issues.couchbase.com/browse/K8S-3617[K8S-3617^]*::

> It's not currently possible to set xref:resource/couchbasecluster.adoc#couchbaseclusters-spec-cluster-indexer-redistributeindexes[`couchbaseclusters.spec.cluster.indexer.redistributeIndexes`] from True to False during a reconciliation.

Reviewer: Automated review comment from Vale and OpenAI using the gpt-4o-mini model:

  ```suggestion
  It's not possible to set xref:resource/couchbasecluster.adoc#couchbaseclusters-spec-cluster-indexer-redistributeindexes[`couchbaseclusters.spec.cluster.indexer.redistributeIndexes`] from True to False during a reconciliation.
  ```
  
  Using these Vale rules:
  * [write-good.Weasel](https://en.wikipedia.org/wiki/Weasel_word) (warning) - &#x27;currently&#x27; is a weasel word!

*https://jira.issues.couchbase.com/browse/K8S-3908[K8S-3908^]*::

Metric `couchbase_operator_memory_under_management_bytes` is incorrectly showing 0.

*https://jira.issues.couchbase.com/browse/K8S-3909[K8S-3909^]*::

Metric `couchbase_operator_cpu_under_management` is incorrectly showing 0.

*https://jira.issues.couchbase.com/browse/K8S-3910[K8S-3910^]*::

> Operator tries to migrate storage backend of buckets even before couchbase cluster is in 7.6.0+.

Reviewer: Automated review comment from Vale and OpenAI using the gpt-4o-mini model:

  ```suggestion
  Operator tries to migrate storage backend of buckets even before Couchbase cluster is in 7.6.0+.
  ```
  
  Using these Vale rules:
  * Vale.Terms (error) - Use &#x27;Couchbase&#x27; instead of &#x27;couchbase&#x27;.

> Operator tries to migrate storage backend of buckets even before couchbase cluster is in 7.6.0+.

Reviewer: Automated review comment from Vale and OpenAI using the gpt-4o-mini model:

  ```suggestion
  Operator tries to migrate storage backend of buckets even before Couchbase cluster is in 7.6.0+.
  ```
  
  Using these Vale rules:
  * Vale.Terms (error) - Use &#x27;Couchbase&#x27; instead of &#x27;couchbase&#x27;.
